src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java

```diff
@@ -332,4 +332,60 @@ private boolean isMatchingByLineAndColumnAndTokenType(DetailAST ast) {
                 && expandedTabColumn(ast) == columnNumber
                 && (tokenType == 0 || tokenType == ast.getType());
     }
+
+    /**
+     * Escape &lt;, &gt;, &amp;, &#39; and &quot; as their entities.
+     *
+     * @param value the value to escape.
+     * @return the escaped value if necessary.
+     */
+    public static String encode(String value) {
```

Comment:
I try to think why we need our special implementation of this 
can you reuse Guava - https://stackoverflow.com/a/52364912/1015848 ?

Reply:
Sir, we need our special implementation because of two reasons: 
1. We use Saxon for decoding xpath, which uses some custom rules like double apos to represent single apos.
2. I am not sure if non-ASCII character required to be encoded, but we are doing it right now. But guava is not able to encode them till now as there documentation states `Currently the escapers provided by this class do not escape any characters outside the ASCII character range`.

Reply:
Please put such reasons to javadoc of method, it is not obvious, let's keep this knowledge written

Reply:
Done.

Reply:
Also, please see [#9247 Comment](https://github.com/checkstyle/checkstyle/pull/9247#discussion_r576550029) .

Reply:
yes all details to javadoc.